test/gentypes.jl

```diff
@@ -313,4 +313,22 @@
         @test length(json_arr) == 2
         @test json_arr[1].menu.header == "SVG Viewer"
     end
+
+    @testset "Missingness" begin
+        json = """
+            [
+                {"a": "w", "b": 5, "c": 9, "d": null},
+                {"a": 3, "b": 4, "c": 2},
+                {"a": 7, "b": 7, "c": 0, "d": 10}
+            ]
+        """
+
+        JSON3.@generatetypes(json)
+        res = JSON3.read(json, Vector{JSONTypes.Root})
+
+        @test res[3].d == 10
+        @test ismissing(res[2].d)
+        @test res[1].d === nothing
```

Comment:
Nit: can we reorder the asserts to be the same order as the input data, and also check for missing and nothing in the same way? Eg,

```jl
@test res[1].d === nothing
@test res[2].d === missing
@test res[3].d == 10
```